src/VisualStudio/Core/Def/Implementation/FindReferences/AbstractFindUsagesCustomColumnDefinition.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.FindUsages
+{
+    /// <summary>
+    /// Implementation of a custom, dynamic column for the Find All References window.
+    /// </summary>
+    internal abstract class AbstractFindUsagesCustomColumnDefinition : TableColumnDefinitionBase
+    {
+        protected AbstractFindUsagesCustomColumnDefinition()
+        {
+            DefaultColumnState = new ColumnState2(Name, isVisible: false, DefaultWidth);
+        }
+
+        public ColumnState2 DefaultColumnState { get; }
+
+        public override bool TryGetFilterItems(ITableEntryHandle entry, out IEnumerable<string> filterItems)
+        {
+            // Determine the constituent strings for the display value in column, which should be used for applying the filter.
+            // For example, if value "Read, Write" is displayed in column for an entry, we will return "Read" and "Write" here,
+            // so filtering based on individual filter terms can be done.
+            if (IsFilterable &&
+                entry.TryGetValue(Name, out var value) &&
+                value is string displayString &&
+                !string.IsNullOrEmpty(displayString))
+            {
+                filterItems = SplitColumnDisplayValue(displayString);
```

Comment:
how often is this called?  once per item in the row?  if so, we may want to cache the split values (and, if you've already done this, ignore this comment).

Reply:
Yep, `SplitColumnDisplayValue` is an abstract method and its implementation caches the mapping from display value to split values.